### PR TITLE
#207 material data filter search

### LINKED ISSUE #207: New query template
Create new query that returns all results for specified field stored in system (e.g. Glass transition temperature values; tan(delta) spectra; etc).
NOTE: filter panel in Data Explorer already achieves this.  Need followup with PIs to determine if additional functionality is desired.

### LINKED ISSUE #212: "Query returned __ results" error
After running a search, the counter "Query returned ___ results" does not return the correct number, especially if submitting subsequent queries. (bug)

### PR DESCRIPTION
**Pull request by tholulomo**
- Refactor BE api to support materials property feature on the UI search.
- Added material property feature as part of search on the UI
- Fix total search count bug.
- Added unit testing.

closes #212 
closes #207

### CODE REVIEW

File: app/src/components/explorer/SearchResultsTable.vue [hunk 1]
```diff
@@ -266,6 +253,9 @@ export default {
     })
   },
   methods: {
+    async loadMaterials (selectedValue) {
```
Comment (aswallace) [inline app/src/components/explorer/SearchResultsTable.vue:256]: If we're changing the table header to properties, we may also want to change these to loadProperties ?

File: app/tests/jest/script/wrapper.js [hunk 2]
```diff
@@ -1,12 +1,15 @@
 import _ from 'lodash'
+import Vuex from 'vuex'
 import VueMaterial from 'vue-material'
 import store from '@/store'
```
Comment (aswallace) [inline app/tests/jest/script/wrapper.js:4]: We also already import store here, is this redundant?

**Review by aswallace — APPROVED** [review]
Looks good. My only comment is that if we're going to change the table header to Properties, we might want to also rename the helper functions. I suspect we'll eventually need a new "Material" table header specifically for metamaterial unit cells

File: app/tests/jest/script/wrapper.js (hunk 2 above)
Comment (tholulomo) [reply app/tests/jest/script/wrapper.js:4]: Yes it is, I added it during testing tweaks for the wrapper and forgot to remove

File: app/src/components/explorer/SearchResultsTable.vue (hunk 1 above)
Comment (tholulomo) [reply app/src/components/explorer/SearchResultsTable.vue:256]: I agree

**Pull request merged**